.github/copilot-instructions.md

```diff
@@ -0,0 +1,122 @@
+   - `unzip -l test-mod.zip | head -20` - Should show control.lua, data.lua, info.json at root
+3. **Mod Metadata Check**: `cat src/info.json` - Verify valid JSON with name, version, title fields
+4. **Core Files Present**: Verify these essential files exist in src/:
+   - `control.lua` (mod runtime logic)
+   - `data.lua` (game data definitions)
+   - `info.json` (mod metadata)
+   - `settings.lua` (mod settings)
+5. **Syntax Validation**: The luacheck command validates Lua syntax across all files
+6. **File Structure**: Check that src/todo/ contains features/ and ui/ subdirectories
+
+## Validation
+- Always run `luacheck src` to check for code issues before making changes
+- Verify mod package structure by creating and examining zip contents
+- Check that any new Lua files follow the existing code patterns
+- Since this is a Factorio mod, full functional testing requires the Factorio game which is not available in this environment
+- ALWAYS run linting after making code changes - the CI will fail if linting errors are introduced
+
+## Common Tasks
+The following are outputs from frequently run commands. Reference them instead of running bash commands when possible to save time.
+
+### Repository Structure
+```
+├── .github/                 # GitHub workflows and configuration
+├── src/                     # Factorio mod source code
+│   ├── control.lua         # Main mod entry point
+│   ├── data.lua            # Game data definitions  
+│   ├── info.json           # Mod metadata and version
+│   ├── settings.lua        # Mod settings
+│   ├── graphics/           # Mod icons and images
+│   ├── locale/             # Translations
+│   └── todo/               # Business logic
+│       ├── features/       # Feature implementations
+│       ├── ui/             # User interface components
+│       ├── todo.lua        # Main todo functionality
+│       ├── helper.lua      # Utility functions
+│       └── logging.lua     # Debug logging
+├── spec/                   # Test files (busted framework)
+├── changelog.txt           # Version history
+├── release.py              # Release automation script
+└── .luacheckrc            # Lua linting configuration
+```
+
+### Package Structure Check
+```bash
+unzip -l todo-list.zip | head -20
+# Expected output: graphics/, control.lua, info.json, locale/, data.lua, 
+# thumbnail.png, settings.lua, todo/ directory with features/ and ui/ subdirs
+```
+
+### Current Linting Status
+```bash
+luacheck src
+# Expected: "Total: 13 warnings / 0 errors in 29 files"
+# This baseline should not regress - no new errors should be introduced
+```
+
+### Key Files to Understand
+- `src/control.lua` - Event handlers and mod initialization
+- `src/todo/todo.lua` - Core todo list functionality  
+- `src/todo/features/` - Individual feature implementations (add_task.lua, edit_task.lua, etc.)
+- `src/todo/ui/` - UI component definitions
+- `src/info.json` - Contains mod version and metadata
```

Comment:
Mention above that this file must be updated for the build pipeline succeed first on the PR and then on the release tag